assets/webpack.config.js

```diff
@@ -7,115 +7,268 @@ const TerserPlugin = require("terser-webpack-plugin");
+                "@babel/preset-typescript",
               ],
-              [
-                "@babel/plugin-proposal-nullish-coalescing-operator",
-                { loose: false },
+              plugins: [
+                "@babel/plugin-proposal-export-default-from",
+                "@babel/plugin-proposal-logical-assignment-operators",
+                ["@babel/plugin-proposal-optional-chaining", { loose: false }],
+                [
+                  "@babel/plugin-proposal-pipeline-operator",
+                  { proposal: "minimal" },
+                ],
+                [
+                  "@babel/plugin-proposal-nullish-coalescing-operator",
+                  { loose: false },
+                ],
+                "@babel/plugin-proposal-do-expressions",
               ],
-              "@babel/plugin-proposal-do-expressions",
-            ],
+            },
           },
         },
+        {
+          enforce: "pre",
+          test: /\.js$/,
+          loader: "source-map-loader",
+        },
+        {
+          test: /\.s?css$/,
+          use: [
+            MiniCssExtractPlugin.loader,
+            {
+              loader: "css-loader",
+            },
+            {
+              loader: "sass-loader",
+            },
+          ],
+        },
+        {
+          test: /\.svg$/i,
+          issuer: /\.[jt]sx?$/,
+          use: ["@svgr/webpack"],
+        },
+        {
+          test: /\.(woff(2)?|ttf|eot)(\?v=\d+\.\d+\.\d+)?$/,
+          use: [
+            {
+              loader: "file-loader",
+              options: {
+                name: "[name].[ext]",
+                outputPath: "fonts/",
+                publicPath: "../fonts/",
+                useRelativePaths: true,
+              },
+            },
+          ],
+        },
+        {
+          test: /\.(png|jpe?g|gif|webp)$/i,
+          use: [
+            {
+              loader: "file-loader",
+              options: {
+                name: "[name].[ext]",
+                outputPath: "/images/",
+                publicPath: "/images/",
+                useRelativePaths: true,
+              },
+            },
+          ],
+        },
+      ],
+    },
+    plugins: [
+      new MiniCssExtractPlugin({ filename: "../css/[name].css" }),
+      new CopyWebpackPlugin({ patterns: [{ from: "static/", to: "../" }] }),
+    ],
+    devtool: "source-map",
+    optimization: {
+      minimizer: [
+        new TerserPlugin({ cache: true, parallel: true, sourceMap: false }),
+        new OptimizeCSSAssetsPlugin({}),
+      ],
+    },
+  },
+  {
+    resolve: {
+      extensions: [".ts", ".tsx", ".js", ".jsx"],
+      alias: {
+        // Please also update the "paths" list in tsconfig.json when you add aliases here!
+        Components: path.resolve(__dirname, "src/components"),
+        Hooks: path.resolve(__dirname, "src/hooks"),
+        Util: path.resolve(__dirname, "src/util"),
+        Constants: path.resolve(__dirname, "src/constants"),
```

Comment:
Since this is a JS file and we can define variables and such, can we extract out common parts of these configs, so it's clear where they differ and we can change shared things in one place?